src/bandit-evaluator.spec.ts

```diff
@@ -0,0 +1,300 @@
+    it('handles no actions', () => {
+      const actionWeights = exposedEvaluator.weighActions({}, 1, 0.1);
+      expect(actionWeights).toStrictEqual({});
+    });
+
+    it('weights a single action at 100%', () => {
+      const scoredActions = { action: 1.23 };
+      const actionWeights = exposedEvaluator.weighActions(scoredActions, 1, 0.1);
+      expect(actionWeights).toStrictEqual({
+        action: 1.0,
+      });
+    });
+
+    it('weighs multiple actions with the same scores', () => {
+      const scoredActions = { action1: 5, action2: 5, action3: 5 };
+      const actionWeights = exposedEvaluator.weighActions(scoredActions, 1, 0.1);
+      expect(Object.keys(actionWeights)).toHaveLength(3);
+      expect(actionWeights.action1).toBeCloseTo(0.33333);
+      expect(actionWeights.action2).toBeCloseTo(0.33333);
+      expect(actionWeights.action3).toBeCloseTo(0.33333);
+    });
+
+    it('weighs multiple actions with different scores', () => {
+      const scoredActions = { action1: 1, action2: 0.5 };
+      const gamma = 10;
+      const actionWeights = exposedEvaluator.weighActions(scoredActions, gamma, 0.1);
+      expect(Object.keys(actionWeights)).toHaveLength(2);
+      expect(actionWeights.action1).toBeCloseTo(0.85714);
+      expect(actionWeights.action2).toBeCloseTo(0.14286);
+    });
+
+    it('responds as expected to changes in gamma', () => {
+      const scoredActions = { action1: 1, action2: 0.5 };
+      const smallGamma = 0.1;
+      const largeGamma = 0.5;
+      const actionWeightsSmallGamma = exposedEvaluator.weighActions(scoredActions, smallGamma, 0.0);
+      const actionWeightsLargeGamma = exposedEvaluator.weighActions(scoredActions, largeGamma, 0.0);
+      // Gamma quantifies the "learning rate" of the FALCON algorithm; with a larger value meaning less learning and smaller more learning
+      // Increasing gamma from low to high, we expect to exploit more and explore less
+      // Thus we expect the higher-scored action's weight to increase and the lower-scored action's weight to decrease
+      expect(actionWeightsLargeGamma.action1).toBeGreaterThan(actionWeightsSmallGamma.action1);
+      expect(actionWeightsLargeGamma.action2).toBeLessThan(actionWeightsSmallGamma.action2);
+    });
+
+    it('applies probability floor', () => {
+      const scoredActions = { action1: 1, action2: 0.5, action3: 0.2 };
+      const gamma = 10;
+      const lowProbabilityFloor = 0.1;
+      const highProbabilityFloor = 0.3;
+      const actionWeightsLowProbabilityFloor = exposedEvaluator.weighActions(
+        scoredActions,
+        gamma,
+        lowProbabilityFloor,
+      );
+      const actionWeightsHighProbabilityFloor = exposedEvaluator.weighActions(
+        scoredActions,
+        gamma,
+        highProbabilityFloor,
+      );
+      // As we increase the probability floor, we expect the lowest scored action's weight to be lifted, the highest scored to be reduced, and the others to the stay the same
```

Comment:
Side note: this is pretty rudimentary, and we should revisit this in the future. I imagine that, ideally, we distribute the total amount of artificially "lifted" weights amongst all those already above the floor in proportion to their unmodified values. /cc @schmit & @giorgiomartini0

Reply:
Agree this is pretty rudimentary, but I think it's fine for now. Realistically, we might want to use probability floors more on the order of 0.01, so there will be less "lifted" weight to redistribute.